### PR TITLE
add task modal

### PR DESCRIPTION
**Pull request by fourwallsstudio**

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -43,13 +44,13 @@ def update_tasks
   def destroy
     @task = current_user.tasks.find(params[:id])
     @task.destroy
-    redirect_to :action => 'list'
+    render json: @task
```
Comment (pseuyi) [inline app/controllers/tasks_controller.rb:47]: this necessary?

File: frontend/components/app.jsx [hunk 2]
```diff
@@ -4,14 +4,16 @@ import { AuthRoute, ProtectedRoute } from '../util/route_util';
...
 
 const App = () => {
   return (
     <section id="app-container">
       <Route exact path="/" component={Landing}/>
       <AuthRoute path="/login" component={SessionModal} />
       <AuthRoute path="/signup" component={SessionModal} />
-      <ProtectedRoute exact path="/home" component={Home} />
+      <ProtectedRoute path="/home" component={Home} />
+      <TaskModal />
```
Comment (pseuyi) [inline frontend/components/app.jsx]: minor - i would expect this render in the `<Home/>` component
also confusing - `SessionModal` is more of a page not a true modal

File: frontend/reducers/modal_reducer.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import * as actions from '../consts/action-types';
+
+const defaultState = {
+  active: false,
```
Comment (pseuyi) [inline frontend/reducers/modal_reducer.js:4]: maybe prefer `active` to refer to a specific modal. assumption is that only one modal can be rendered in the app?

**Review by pseuyi — APPROVED** [review]
some minor comments. most concerned about scaling the modal reducer - see comment about modal.active

**Comment by pseuyi** [comment]
lgtm~

File: app/controllers/tasks_controller.rb (hunk 1 above)
Comment (fourwallsstudio) [reply app/controllers/tasks_controller.rb:47]: ya, the redirect throws errors.
we do have the option:
1) render with task a remove single task from front end store.
2) render all tasks from back end after delete

**Pull request merged**